HHS/Head-Start-TTADP
# TTAHUB-164,206,263,279: rename legacy reasons, implement new bar graph filters, add role filter, and alphabetize multiselects

## Description of change

- Rename legacy Reasons to the new standard naming convention.
- Multi selects alphabetize input as it is entered. Multi select data is also sorted alphabetically on the approved report view, the review/draft report view, and the CSV exports.
- The filter dropdown on /activity-reports now has an additional option - role. Users can select either role "contains" or "does not contain" from a predetermined set of options. A corresponding scope has been added to the backend.
- Implemented new design for the specialists filter (now a set of checkboxes) and the sorting (the options now match the original design, and the apply filters button is within the dropdown menu).


## How to test

Run the job against a db that has legacy reasons. All legacy reasons should be updated while new reasons should be the same.

If an AR has three reasons that now fall under a single reason. The new reason should only appear once.

Create a new report. Enter multi select data in non-alpha order and see how it is displayed. View prior activity reports and note that all the multi-select fields have been alphabetized.

Navigate to /activity-reports. Create a filter for role. Try a few different options, both filtering by and filtering out reports with certain roles.

Visit the dashboard. Scroll to the bar graph at the bottom. Observe that the controls for sorting and filtering by specialist are different - the apply filters is confined to each box now. Test them out. Watch the data filter and the graph redraw.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-164
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-206
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-263
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-279


## Checklists

### Every PR


~- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)~
~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

~- [ ] Staging smoke test completed~

### After merge/deploy
